File: app/src/main/java/com/maciejkozlowski/databases/greendao/CitiesLoaderDao.java
package com.maciejkozlowski.databases.greendao;


import android.content.Context;

import com.maciejkozlowski.databases.BaseLoader;
import com.maciejkozlowski.databases.Timings;
import com.maciejkozlowski.databases.results.ResultSet;

import java.util.List;

/**
 * Created by Maciej Kozłowski on 01.05.17.
 */
public class CitiesLoaderDao extends BaseLoader<CityDao> {

    public static final String TAG = "dao";

    public void execute(Context context, ResultSet resultSet, CityDaoDao cityDaoDao, int size) {
        List<CityDao> cities = readFromFile(context, Companion.getCITIES_CSV(), size);

        getLogger().start();
        cityDaoDao.insertInTx(cities);
        getLogger().logTime(resultSet.getCreating(), Companion.getINSERT_CITIES(), size);

        getLogger().start();
        List<CityDao> cityDaos = cityDaoDao.loadAll();
        getLogger().logTime(resultSet.getReading(), Companion.getREAD_CITIES(), size);

        getLogger().start();
        for (int i = 0; i < cityDaos.size(); i++) {
            cityDaos.get(i).setName(String.valueOf(i));
        }
        cityDaoDao.updateInTx(cityDaos);
        getLogger().logTime(resultSet.getUpdating(), Companion.getUPDATE_CITIES(), size);

        getLogger().start();
        cityDaoDao.deleteAll();
        getLogger().logTime(resultSet.getDeleting(), Companion.getDELETE_CITIES(), size);
    }

    @Override
    protected CityDao create(Long id, String name, Double latitude, Double longitude) {
        return new CityDao(id, name, latitude, longitude);
    }

    @Override
    protected Timings createTimingLogger() {
        return new Timings(TAG);
    }
}

File: app/src/main/java/com/maciejkozlowski/databases/greendao/CitiesLoaderDao.kt
package com.maciejkozlowski.databases.greendao


import android.content.Context

import com.maciejkozlowski.databases.BaseLoader
import com.maciejkozlowski.databases.Timings
import com.maciejkozlowski.databases.results.ResultSet

/**
 * Created by Maciej Kozłowski on 01.05.17.
 */
class CitiesLoaderDao : BaseLoader<CityDao>() {

    fun execute(context: Context, resultSet: ResultSet, cityDaoDao: CityDaoDao, size: Int) {
        val cities = readFromFile(context, CITIES_CSV, size)

        logger.start()
        cityDaoDao.insertInTx(cities)
        logger.logTime(resultSet.creating, INSERT_CITIES, size)

        logger.start()
        val cityDaos = cityDaoDao.loadAll()
        logger.logTime(resultSet.reading, READ_CITIES, size)

        logger.start()
        for (i in cityDaos.indices) {
            cityDaos[i].name = i.toString()
        }
        cityDaoDao.updateInTx(cityDaos)
        logger.logTime(resultSet.updating, UPDATE_CITIES, size)

        logger.start()
        cityDaoDao.deleteAll()
        logger.logTime(resultSet.deleting, DELETE_CITIES, size)
    }

    override fun create(id: Long?, name: String, latitude: Double?, longitude: Double?): CityDao {
        return CityDao(id!!, name, latitude!!, longitude!!)
    }

    override fun createTimingLogger(): Timings {
        return Timings(TAG)
    }

    companion object {

        const val TAG = "dao"
    }
}
